Harvest.Web/ClientApp/src/Requests/Crops.tsx

```diff
@@ -40,8 +43,29 @@ export const Crops = (props: Props) => {
     }
   };
 
+  const onSearch = async (query?: string) => {
+    setIsSearchLoading(true);
+
+    const response = await fetch(
+      `/Crop/Search?type=${props.cropType}${(query && `&query=${query}`) || ""}`
+    );
+
+    if (response.ok) {
+      if (response.status === 204) {
+        getIsMounted() && setOptions([]); // no content means no match
+      } else {
+        const crops: string[] = (await response.json()).map(
+          (c: { name: string }) => c.name
+        );
+
+        getIsMounted() && setOptions(crops);
+      }
+    }
+    getIsMounted() && setIsSearchLoading(false);
+  };
+
   return (
-    <Typeahead
+    <Typeahead<string>
```

Comment:
is this newly required?  never seen this syntax inside component before, looks different

Reply:
No, but I didn't like that options were being stored as type `string | { label: string }`. Which reminds me, I forgot to remove that union from the onSelect parameter.

Reply:
Nevermind. Just realized that the funky union type it was showing was only due to the data being passed in. Making the generic type explicit is redundant, so I removed it.